### PR TITLE
feat: replace axios with fetch in js client sdk

### PR DESCRIPTION
**Pull request by elliotCamblor**
- Swap out axios for fetch
- Switch to using the `server-request` lib which adds functionality for fetch like exponential backoff, retires, and error handling
- Duplicate the aforementioned lib INTO the js sdk src folder because the nx:rollup build process can't handle the import

### CODE REVIEW

File: sdk/js/rollup.config.mjs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            file: 'dist/sdk/js/index.esm.js',
+            format: 'esm',
+        }
+    ],
+    plugins: [
+        // Other plugins...
+        resolve(), // Make sure resolve plugin is used
+        alias({
+            entries: [
+                { find: '@devcycle/server-request', replacement: 'lib/shared/server-request/src/index.ts' },
```
Comment (ajwootto) [inline sdk/js/rollup.config.mjs:21]: this is really sketchy...and seems like it would definitely cause issues with Nx.

Comment (elliotCamblor) [reply sdk/js/rollup.config.mjs:21]: yeah, this isnt supposed to be here, didnt mean to push this config up. Apparently overriding the rollup config is the only way to fix the issue im facing tho so Im not sure what to do

File: sdk/js/.eslintrc.json [hunk 2]
```diff
@@ -9,7 +9,11 @@
                 "@nx/dependency-checks": [
                     "error",
                     {
-                        "ignoredDependencies": []
+                        "ignoredDependencies": [
+                            "fetch-retry",
+                            "cross-fetch",
+                            "@devcycle/server-request"
```
Comment (ajwootto) [inline sdk/js/.eslintrc.json:15]: does this still need to be here?

File: sdk/js/.eslintrc.json [hunk 3]
```diff
@@ -9,7 +9,11 @@
                 "@nx/dependency-checks": [
                     "error",
                     {
-                        "ignoredDependencies": []
+                        "ignoredDependencies": [
+                            "fetch-retry",
```
Comment (ajwootto) [inline sdk/js/.eslintrc.json:13]: you shouldnt have to ignore any of these now I think it should pick up their usage

Comment (elliotCamblor) [reply sdk/js/.eslintrc.json:13]: probably right, a lot of this got lost in translation with my 1.4 billion iterations to fix the build. Ill test it out

File: sdk/js/project.json [hunk 4]
```diff
@@ -46,7 +46,7 @@
                 "entryFile": "sdk/js/src/index.ts",
                 "compiler": "tsc",
                 "format": ["esm", "cjs"],
-                "external": "all",
+                "external": ["shared-types"],
```
Comment (elliotCamblor) [inline sdk/js/project.json:49]: revert this

File: lib/shared/server-request/src/request.ts [hunk 5]
```diff
@@ -1,3 +1,6 @@
+// NOTE: This file is duplicated in "sdk/js/src/RequestUtils" because nx:rollup cant build non-external dependencies
+// from outside the root directory
```
Comment (elliotCamblor) [inline lib/shared/server-request/src/request.ts:2]: github issue outlining this problem https://github.com/nrwl/nx/issues/10395

Comment (jonathannorris) [reply lib/shared/server-request/src/request.ts:2]: can you add this as a code-comment.

File: lib/shared/server-request/src/request.ts [hunk 6]
```diff
@@ -88,6 +93,26 @@ export async function post(
...
+    }
+
+    const res = await _fetch(url, {
+        ...config,
+        headers: patchHeaders,
+        method: 'PATCH',
+    })
+
+    return handleResponse(res)
+}
```
Comment (jonathannorris) [inline lib/shared/server-request/src/request.ts:115]:
```suggestion
}

```

File: sdk/js/src/Request.ts [hunk 7]
```diff
@@ -103,18 +56,15 @@ export const getConfigJson = async (
         queryParams.toString()
 
     try {
-        const res = await get(url)
-        return res.data
-    } catch (ex: any) {
-        const errorString = JSON.stringify(ex?.response?.data?.data?.errors)
+        const res = await getWithTimeout(url, requestConfig, 5000)
```
Comment (jonathannorris) [inline sdk/js/src/Request.ts:59]: Was 5 seconds the timeout before? or is this new?

Comment (elliotCamblor) [reply sdk/js/src/Request.ts:59]: The timeout was 5 seconds before, it was just being set on the axios client https://github.com/DevCycleHQ/js-sdks/pull/739/files#diff-b985bf038e49b8ee77cff70485df18db4d9fd1d6ff6992dd90a1038f9b547393L8

**Review by jonathannorris — APPROVED** [review]

**Pull request merged**